extensions/amp-bind/0.1/amp-state.js

```diff
@@ -50,9 +50,27 @@ export class AmpState extends AMP.BaseElement {
     toggle(this.element, /* opt_display */ false);
     this.element.setAttribute('aria-hidden', 'true');
 
-    // Don't parse or fetch in prerender mode.
-    const viewer = Services.viewerForDoc(this.getAmpDoc());
-    viewer.whenFirstVisible().then(() => this.initialize_());
+    const {element} = this;
+    if (element.hasAttribute('overridable')) {
+      Services.bindForDocOrNull(element).then(bind => {
+        devAssert(bind);
+        bind.addOverridableKey(element.getAttribute('id'));
+      });
+    }
+    // Parse child <script> tag and/or fetch JSON from `src` attribute.
+    // The latter is allowed to overwrite the former.
+    const {children} = element;
+    if (children.length > 0) {
+      // Bind relies on this happening synchronously in buildCallback().
+      this.parseAndUpdate_();
+    }
+    if (this.element.hasAttribute('src')) {
+      this.fetchAndUpdate_(/* isInit */ true);
+    }
+
+    this.registerAction('refresh', () => {
```

Comment:
I should have made this change when I added refresh support for amp-state but we should only register the action if the element has a src attribute. We should move this block in the if statement above.

    if (this.element.hasAttribute('src')) {
      this.fetchAndUpdate_(/* isInit */ true);
      this.registerAction('refresh', ...
    }

Reply:
Good call. Though it's possible for the element to not have a `src` attribute at buildCallback but have one later (via amp-bind), so I added a `userAssert` here instead.